### ISSUE TITLE
Regression in typescript generics formatting

### ISSUE THREAD

**Issue by kadircet** [llvm/llvm-project#108536]
actual formatting (file name `a.ts`):
```
foo = new Bar < (id: int) => X | Y > ();
```

expected formatting:
```
foo = new Bar<(id: int) => X | Y>();
```

**Comment by kadircet** [comment]
culprit seems to be 73c961a3345c697f40e2148318f34f5f347701c1, cc @owenca, @rymiel

**Comment by kadircet** [comment]
looks like this was also cherry-picked into 19.x release https://github.com/llvm/llvm-project/pull/101112

**Comment by kadircet** [comment]
as mentioned in https://github.com/llvm/llvm-project/pull/108671#discussion_r1762725421, this is not specific to js:
```cpp
constexpr auto foo() { return FixedInt < N | M > (foo); }
```

**Comment by owenca** [comment]
See https://github.com/llvm/llvm-project/pull/108671#discussion_r1772483173.

**Comment by owenca** [comment]
> as mentioned in [#108671 (comment)](https://github.com/llvm/llvm-project/pull/108671#discussion_r1762725421), this is not specific to js:
> 
> ```c++
> constexpr auto foo() { return FixedInt < N | M > (foo); }
> ```

Split to #109912.

**Comment by owenca** [comment]
> actual formatting (file name `a.ts`):
> 
> ```
> foo = new Bar < (id: int) => X | Y > ();
> ```
> 
> expected formatting:
> 
> ```
> foo = new Bar<(id: int) => X | Y>();
> ```

This was fixed by #110408.

**Comment by owenca** [comment]
> as mentioned in [#108671 (comment)](https://github.com/llvm/llvm-project/pull/108671#discussion_r1762725421), this is not specific to js:
> 
> constexpr auto foo() { return FixedInt < N | M > (foo); }

See #124438.
